gossipd/gossip.c

```diff
@@ -2394,6 +2394,11 @@ static void gossip_disable_channel(struct routing_state *rstate, struct chan *ch
 		    "Unable to parse previously accepted channel_update");
 	}
 
+	/* Avoid sending gratuitous disable messages, e.g., on close and
+	 * subsequent disconnect */
+	if (flags & ROUTING_FLAGS_DISABLED)
+		return;
+
```

Comment:
I have the same patch in my queue, though I genericised it (so any "no-change" update gets suppressed except keepalives), but this is still a good change.